### PR TITLE
ci: twister: blackbox: do not fetch hal modules

### PR DESCRIPTION
**Pull request by nashif**
Do not fetch optional modules while running this action.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

File: .github/workflows/twister_tests_blackbox.yml [hunk 1]
```diff
@@ -43,6 +43,7 @@ jobs:
         echo "$HOME/.local/bin" >> $GITHUB_PATH
 
         west init -l . || true
+        west config manifest.group-filter -- +ci,-optional
```
Comment (fabiobaltieri) [inline .github/workflows/twister_tests_blackbox.yml:46]: `-optional` is redundant right? that should be the default

not asking to change it just trying to understand

**Review by fabiobaltieri — DISMISSED** [review]

File: .github/workflows/twister_tests_blackbox.yml (hunk 1 above)
Comment (nashif) [reply .github/workflows/twister_tests_blackbox.yml:46]: > `-optional` is redundant right? that should be the default

is it the default? defined where? AFAIK, without filters, everything is fetched by default. see `.github/workflows/compliance.yml` for example.

Either way, this fix is not enough, we have a "bigger" problem....

Comment (fabiobaltieri) [reply .github/workflows/twister_tests_blackbox.yml:46]: > is it the default? defined where?

Here? https://github.com/zephyrproject-rtos/zephyr/blob/main/west.yml#L27 if that's not effective you want to filter out babblesim as well

Comment (nashif) [reply .github/workflows/twister_tests_blackbox.yml:46]: oh yeah, you are right. Then I have to find a more directy approach now :(

**Review by fabiobaltieri — APPROVED** [review]

**Review by LukaszMrugala — APPROVED** [review]

**Pull request merged**